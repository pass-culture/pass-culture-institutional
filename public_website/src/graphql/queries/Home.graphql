query Home(
  $filters: ResourceFiltersInput
  $pagination: PaginationArg
  $sort: [String]
) {
  home {
    aboutSection {
      ...ComponentBlockCenteredText
    }
    CTASection {
      ...ComponentBlockPushCta
    }
    socialMediaSection {
      ...ComponentBlockSocialMedia
    }
    latestStudies {
      ...ComponentBlockLatestNews
    }
    eligibilitySection {
      ...ComponentHomeEligibilitySection
    }
    heroSection {
      ...ComponentHomeHeroSection
    }
    recommendationsSection {
      ...ComponentHomeRecommendationsSection
    }
    seo {
      ...ComponentSharedSeo
    }
  }
  resources(filters: $filters, pagination: $pagination, sort: $sort) {
    ...Resource
  }
}
