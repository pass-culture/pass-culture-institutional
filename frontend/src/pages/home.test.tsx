import React from 'react'
import { describe, expect, it } from 'vitest'
import { axe } from 'vitest-axe'

import Home from './index'
import { render } from '@/tests'

describe('Home page', () => {
  it('should pass axe accessibility tests', async () => {
    const { container } = render(<Home />)

    expect(await axe(container)).toHaveNoViolations()
  })
})
