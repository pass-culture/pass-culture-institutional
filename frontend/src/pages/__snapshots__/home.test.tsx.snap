// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Home page > should render the page 1`] = `
<div>
  <div
    class="sc-guDMob kjJDxW"
  >
    <main
      class="sc-beyTiQ AfwsS"
    >
      <h1
        class="sc-iBdoyZ kqGUgJ"
      >
        Welcome to
         
        <a
          class="sc-dmyDmy dPmvRl"
          href="https://nextjs.org"
        >
          Next.js!
        </a>
      </h1>
      <p
        class="sc-fsYfxw gRDVXU"
      >
        Get started by editing
        <code
          class="sc-gLLuHO fhjfjc"
        >
          pages/index.tsx
        </code>
      </p>
      <div>
        <input
          data-testid="checkbox-acceptTerms"
          id="acceptTerms"
          type="checkbox"
        />
        <label
          for="acceptTerms"
        >
          Checkbox à cocher
        </label>
      </div>
      <div
        class="sc-hLQTFJ hHliZR"
      >
        <div
          class="sc-eDLKEg gZPZKf"
        >
          <a
            class="sc-jTQDJr fAvfvS"
            href="/about"
          >
            About →
          </a>
        </div>
      </div>
    </main>
  </div>
</div>
`;
